packages/fields/src/types/Relationship/views/Controller.js

```diff
@@ -28,26 +28,43 @@ export default class RelationshipController extends FieldController {
     return `${this.getFilterLabel({ label })}: "${value}"`;
   };
 
-  // TODO: FIXME: This should be `set`, not `connect`
-  buildRelateToOneInput = ({ id }) => ({ connect: { id } });
-  buildRelateToManyInput = data => ({ connect: data.map(({ id }) => ({ id })) });
+  deserialize = data => {
+    // this is probably not a great solution
+    // ideally, we should have a set option in the mutation
+    this.lastSavedState = this.config.many
+      ? (data[this.path] || []).map(x => x.id)
+      : data[this.path]
+      ? data[this.path].id
+      : null;
+    return data[this.path];
+  };
 
   serialize = data => {
     const { path } = this;
     const { many } = this.config;
 
-    if (!data[path]) {
-      return many ? { disconnectAll: true } : null;
-    }
+    let value = data[path];
 
     if (many) {
-      if (!Array.isArray(data[path]) || !data[path].length) {
-        return { disconnectAll: true };
+      let ids = [];
+      if (Array.isArray(value)) {
+        ids = value.map(x => x.id);
       }
-      return this.buildRelateToManyInput(data[path], path);
+      return {
+        connect: ids.map(id => ({ id })),
```

Comment:
Looks like we're instructing the resolver to `connect` each id present in the field in the Admin UI _each time the item is saved_, while only removing ids that are no longer present. I've realised that how this API works is a bit unclear; I expect this would create _duplicate connections_ which could explain #1353

If that's the case, we could ensure we only connect IDs that aren't present in `lastSavedState`...

We should also figure out what to do with multiple instances of the same value in a relationship field. AFAIK that's actually potentially a valid use case, and would (should?) be supported by the underlying database adapters, but at the moment react-select wouldn't allow you to select the same value more than once without some tweaking.

In the future this might be something we want to introduce an option for, like `allowDuplicates` or something... just to make the behaviour really explicit at a schema level; and then if that's not enabled, we could ensure duplicate values are stripped at the resolver level.